streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java

```diff
@@ -137,6 +137,15 @@ public void shouldThrowProcessorStateExceptionOnOpeningReadOnlyDir() {
 
         try {
             rocksDBStore.openDB(tmpContext);
+	    // For non-root user execution, exception will be thrown.
+            // For root user execution, no exception will be thrown, it will proceed ahead and reach here.
+            // Currently only checking Unix, for other OS need to check(for now it will work as before).
+            final String OS = System.getProperty("os.name").toLowerCase();
+            final boolean isUnix = OS.indexOf("nix") >= 0 || OS.indexOf("nux") >= 0 || OS.indexOf("aix") > 0;
+            if (isUnix) {
+	      // For root user execution on Unix, throw ProcessorStateException.
+       	      throw new ProcessorStateException("root user on Unix can open ReadOnlyDir");
```

Comment:
This seems to defeat the purpose...

If we really want to skip this test in this environment, we should rather put it in the beginning and do
```
if (isUnix) {
  return;
}
```

Reply:
Done, please review!
https://github.com/apache/kafka/pull/6020/commits/6f3e88c3f8a3ec2ecf65125210873eccdb6b8617